### PR TITLE
Architecture: Resource Manager Cleanup

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radix-engine/src/model/resources/executables/resource_manager.rs [hunk 1]
```diff
@@ -72,139 +64,104 @@ impl Executor for ResourceManagerBucketBurnInvocation {
...
-                }
-                resource_manager.total_supply = entries.len().into();
-                let ids = entries.into_keys().collect();
-                let container = Resource::new_non_fungible(resource_address, ids, id_type);
-                let node_id = api.allocate_node_id(RENodeType::Bucket)?;
-                api.create_node(node_id, RENode::Bucket(BucketSubstate::new(container)))?;
-                let bucket_id = node_id.into();
-                Some(Bucket(bucket_id))
-            } else {
+    let bucket = {
```
Comment (0xOmarA) [inline radix-engine/src/model/resources/executables/resource_manager.rs:89]: Why not create the resource manager node first and then do a `ResourceManagerMintNonFungibleInvocation` to mint the non-fungible tokens? Is this potentionally something to do with auth?

**Review by 0xOmarA — APPROVED** [review]
LGTM, left a few minor comments

File: radix-engine/src/model/resources/executables/resource_manager.rs (hunk 1 above)
Comment (talekhinezh) [reply radix-engine/src/model/resources/executables/resource_manager.rs:89]: Yup, one of the reasons, as well as not wanting to make another invocation to setup the initial state.

File: scrypto/src/resource/resource_builder.rs [hunk 2]
```diff
@@ -642,30 +634,38 @@ impl NonFungibleResourceWithAuthBuilder {
...
+            )
+            .unwrap();
+
+        bucket
     }
 
-    fn build(mut self, mint_params: Option<MintParams>) -> (ResourceAddress, Option<Bucket>) {
+    /// Creates resource with no initial supply.
+    pub fn no_initial_supply(mut self) -> ResourceAddress {
         if !self.authorization.contains_key(&Withdraw) {
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply scrypto/src/resource/resource_builder.rs:659]: This is a good point. Will remove.

**Pull request merged**